### ISSUE TITLE
Craft files?

### ISSUE THREAD

**Issue by taterkerman** [shadowmage45/SSTULabs#455]
I'm unsure how accurate these are, particularly the Saturn V's use of the ISDCs, but here are a couple:

[Apollo rescale 1_3.craft.zip](https://github.com/shadowmage45/SSTULabs/files/1036721/Apollo.rescale.1_3.craft.zip)

[Soyuz 1_3.craft.zip](https://github.com/shadowmage45/SSTULabs/files/1036722/Soyuz.1_3.craft.zip)

Having a couple stock test craft might not be a bad idea, because tests using them would be consistent.

**Comment by Jimbodiah** [comment]
Check my repository topic on the forum. As soon as the next release comes out I will update all my craft files. I'd be happy to add your rescales if you like?

**Comment by taterkerman** [comment]
The "rescale" name is my own internal name because I messed with it from the first one I did, trying to make the dimensions 64% of RL.

Any repository would be useful, and I suppose at some point example craft files for the mod itself might be nice. I have simply deleted the stock KSP craft files, and put a few of my own in, as I cannot stand the example craft in KSP (even when using stock parts, I make nothing that looks like the craft files provided).

**Comment by shadowmage45** [comment]
Hmmm....   any opposition (or interest) to setting up an actual 'Repository' for SSTU-based craft, as a secondary github repository?  This could make it a bit more 'official', while still separating the need for updating/testing craft with every full SSTU release.

Could give admin rights to whomever(plural even) was interested in running the show, but could also try and open it up a bit more to community contributions through PRs for updates/fixes/new designs.

Personally I'm fine with Jimbodiah's current setup and distribution and have directed quite a few people towards it who were asking for craft files.  Moving things to Github really doesn't add that much/bring anything new except perhaps making collaboration a bit easier (but only then if you are comfortable with branches/PRs/use of other git-related features);  so entirely up to you guys, just throwing it out there as an option : )

**Comment by Jimbodiah** [comment]
It's your baby... your call. If people can chip in and PR new designs, that is always cool. 

I've been thinking of adding some ships that are geared towards contract missions, stock and RSS, instead of just the real-life replicas. It's just that KSP keeps coming out with new versions too often :)  A github rep might offer some extra directions.

**Comment by Jimbodiah** [comment]
Maybe just go for it? If tater is interested, maybe make him and myself able to admin the repository?

**Comment by taterkerman** [comment]
If jimbodiah already has such a thing going, we could also participate there. I also have a Mercury-Atlas attempt (that drops 2 booster engines, leaving the sustainer---which reminds me that an explicit half-stage mount might be a cool thing... perhaps it could be a variant decoupler that allows a mount, and includes crossfeed and a hollow collider?

**Comment by Jimbodiah** [comment]
I have a forum topic with craft downloads, nothing on github. We can move our craft files to the github repository and manage them there. This way other people could add their own crafts via a PR (or we could do it for them). I suggest Mage stay head-honcho and just delegate the daily chore to us.

**Comment by shadowmage45** [comment]
That was my main reasoning behind the suggestion/idea -- allow a bit more ease of access for community contributions and also de-centralize the burden of management (by having multiple people who can accept PR's and publish new releases).

Could even set up the repository with various smaller themed 'craft packs' if desired, allowing for greater end-user control over what craft they download/install.

Could still have links to the downloads right from Jimbodiah's forum page (possibly even direct links to his packs, if that is how the packs get set up).  The only thing that would need to change there would be the destination of the links on the forum page.  Of course I can link to it (or some of the packs) from the main SSTU page as well.

As it sounds like you guys are both on board (at least for further investigation), I'll work on getting the base repository set up in the next day or two, and add you both as admins for it.  Should be easy from there to upload the initial craft files / releases.  If nothing ends up coming out of it, no worries, only takes a few minutes for the initial setup :)

**Comment by taterkerman** [comment]
Cool. I am also just fine with non-replicas, I tend to make replicas mostly for my rescale games as a benchmark (though I also love the Eyes Turned Skywards concept).

Novel craft combinations are always pretty cool to mess around with.

**Comment by Jimbodiah** [comment]
I have a bunch of patches for SSTU to make my own specialized parts. If you like ETS stuff, we could perhaps work on a patch for such parts (nothing to do with the craft repository). Maybe even José could join in. I still want to figure out how to make a single part ATV-ish craft and need to finish my 2.5m 3-man Orion CM/SM/BPC.

**Comment by Jimbodiah** [comment]
Playing with Kronal Vessel Viewer, maybe nice for some screenshots for the new repository :)

My 1.3.0 stock scale version of a tourist contract run to Mun + Minmus. Stock is soooo easy :)

![sstu_kvv1](https://cloud.githubusercontent.com/assets/16107012/26657346/95052a4e-4663-11e7-8f70-b2f0e5d83dff.png)

**Comment by taterkerman** [comment]
[M atlas 2.craft.zip](https://github.com/shadowmage45/SSTULabs/files/1043791/M.atlas.2.craft.zip)

This is the Mercury I have shown on the forum a few times. Shed the 1/2 stage when the TWR is about 2.6. Because of the nature of the decouple/collider kludge I used, I have had a few accidents, but it usually works. It's a little overbuilt, this one is scaled down from a slightly taller version I can use in 3.2X.

**Comment by Jimbodiah** [comment]
I've added them to the page. 

I need to make better screenshots somehow though. Kronal vesselviewer cuts off the sides so that is not an option, and the ksp thumbnails are... well... $#!^   I'll be adding more career builds later today, should give a good assortment for people to start off with SSTU.

**Comment by shadowmage45** [comment]
Just got the repository created, very basic setup so far with one folder for each contributor (can work on actual distribution/organization setup as we go).

https://github.com/shadowmage45/SSTULabsCraft

Have added you both as admins; you should have full push/pull/read/write access.

Please let me know if you run into any questions on how to add files or get things setup/moving.

Throughout the week I'll work on putting together some build script(s) that we can each use to pack things up for releases, I'll also spend a bit of time thinking on how to best manage the releases, and what kind of organization scheme things should be put into.

**Comment by Jimbodiah** [comment]
Perhaps we should work out some kind of folder structure?

Parameters we have to work with:  Stock/RSS, Rocket/Lander/Station, Specialized career ships

**Comment by taterkerman** [comment]
Sounds reasonable. 

Within a given scale (RSS/stock/whatever), perhaps also replica vs novel craft.

**Comment by shadowmage45** [comment]
Yeah, I think the first organization layer will likely split craft files by scale/purpose:
1. Kerbal/stock scaled craft. ~64% of real scale for pods, and/or setup with performance to be balanced in stock system with <4k-dV-to-orbit requirements.  This one might be sub-divided between '64% scaled replicas' and 'craft for stock system' (or perhaps put 64% scaled replicas into a third 'main' category'?)  
2.  RO / 100% scaled craft.  Real world scales for pods/etc, balanced to be usable in an RO-type system with ~9k-dV-to-orbit requirements.

From there... RO/100% craft would likely be able to be setup by the real-world rocket they are based on.  Same with the 64% replica craft.

The kerbal-balanced craft... are a bit harder to pin down.  Probably should be organized based on purpose -- generic lifters, crewed missions, lander launches, etc.  

Certainly open to debate/discussion on how all this should be set up;  I don't want to mandate any rules or organizational schemes, and will more than likely be fine with whatever you guys want to go with.

Edit:  Regarding the RO/100% craft -- it almost seems like they would need to be distributed with a set of patches to rescale/rebalance the parts (when not used with RO, as it has its own patches).  Not really sure how to best handle that.

Also -- thematic balance patches might also be included in other 'craft packs' if needed (e.g. for a 6.4x rescale and corresponding craft).

**Comment by Jimbodiah** [comment]
I think RO has it's own patches to adjust masses and actual scales. Best not to make additional patches, especially since I don't use RO to play RSS. The craft files could be used for other rescale mods out there, I know there is a 2x, 3.2x and a 6.4x out there somewhere. 

BTW realistically looking rockets work fine in Stock, they just have way too much dV for the real world missions they flew.

**Comment by taterkerman** [comment]
Yeah, that is my experience in stock with SSTU (and stock, for that matter). Stock system requires sub craft for efficiency.

3.2 and stock craft tends to result in entirely realistic looking rockets given the stock scaling of engines, etc.

4X and up requires a little tweaking. I can fly in 4X and 6.4X with stock parts, but the craft start looking bigger than real craft (or the same but with added strap on boosters).

For the sake of this, perhaps we should have a standard for testing 4X, 6.4X, etc (if it involves patching engine mass/thrust/whatever).

**Comment by Jimbodiah** [comment]
Not sure if testing all the different scales is worth it, this will create a ton of extra ships. The beauty of SSTU is that you can easily make tanks smaller on RSS ships to tune it for your own game play. The difference between 3.2 and 4x is not worth seperate craft file imho, and 6.4x can use RSS scale ships. 

I play Stock/SSRSS and RSS, so can test those ships in normal game play, if you could also do one of the popular rescales then I leave that up to you.


Sidenote: I am making a patch to tune isp/thrust to get realistic looking ships in SSRSS/Stock at the moment; just for my own amusement  :)  Basically a RSS ship used in stock that will behave comparable to the RSS variant; i.e. a Delta IV Heavy that will only get into LEO instead of going to Mars.
